src/cls/rbd/cls_rbd.cc

```diff
@@ -772,11 +772,6 @@ int set_stripe_unit_count(cls_method_context_t hctx, bufferlist *in, bufferlist
     CLS_ERR("failed to read the order off of disk: %s", cpp_strerror(r).c_str());
     return r;
   }
-  if ((1ull << order) % stripe_unit || stripe_unit > (1ull << order)) {
```

Comment:
We should keep this check too, since librbd will still use it when not called from /usr/bin/rbd